### DISCUSSION TITLE
How to use $fetch in ssr with local url

### DISCUSSION THREAD

**Discussion by ToBinio** [nuxt/nuxt#26614] [Questions] [answered]
I am trying to fetch some data from one of my nitro endpoints via `$fetch`. But this fails in ssr with 

```
[GET] "/api/test": <no response> Failed to parse URL from /api/test
```

on the client, it works as expected.
here's some example code:

```vue
<script setup lang="ts">
import {$fetch} from "ofetch";

let val = ref("");

// This works normal
onMounted(async () => {
  val.value = await $fetch("/api/test");
})

//this fails with - [GET] "/api/test": <no response> Failed to parse URL from /api/test
val.value = await $fetch("/api/test");
</script>

<template>
  <div>
    {{ val }}
  </div>
</template>
```

with `useFetch` it also works as expected. 
The reason I cant use `useFetch` for my case is that I do not always need to fetch something and having a useFetch inside of a computed valued with an if, simply sounds wrong. 

How should I handle this or is this a bug I should report?

thanks in advance!

**Comment by TheAlexLichter** [comment, marked as answer]
you shouldn't import `$fetch` from `ofetch` as it is auto-imported with a default base URL:

https://github.com/nuxt/nuxt/blob/bcca7aa71e2bcf8e86965275974108cbd6e8f116/packages/nuxt/src/core/templates.ts#L356-L369).

Then, relative URLs will work as expected

**Reply by ToBinio** [reply]
Thanks a lot...
I slowly hate WebStorm... it always auto imports it when I use it.
